api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -49,6 +50,18 @@ message UdpProxyConfig {
     }
   }
 
+  // Configuration for UDP session filters.
+  message SessionFilter {
+    // The name of the filter configuration.
+    string name = 1 [(validate.rules).string = {min_len: 1}];
+
+    oneof config_type {
+      // Filter specific configuration which depends on the filter being
+      // instantiated. See the supported filters for further documentation.
+      google.protobuf.Any typed_config = 2;
+    }
+  }
```

Comment:
I would replace this with `TypedExtensionConfig`, which is better for new cases and avoids duplication.

Reply:
I would like to add ECDS support in the future for session filters, which is not supported by the ``TypedExtensionConfig`` (has no ``config_discovery`` field and doesn't have ``oneof``). I kind of hesitate to use this and then risk that it would cause problems when I'll want to add ECDS. wdyt?

Reply:
Sure sounds good.